File: src/main/java/com/zerozero/store/infrastructure/mongodb/Store.java
package com.zerozero.store.infrastructure.mongodb;

import com.zerozero.core.util.GeoJsonConverter;
import lombok.*;
import org.springframework.data.mongodb.core.geo.GeoJsonPoint;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.UUID;

@ToString
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "store")
public class Store {

    private UUID storeId;

    private String kakaoId;

    private String name;

    private String category;

    private String phone;

    private String address;

    private String roadAddress;

    private String longitude;

    private String latitude;

    private GeoJsonPoint location;

    private String placeUrl;

    private boolean status;

    private UUID userId;

    public static Store of(com.zerozero.store.domain.model.Store store) {
        return Store.builder()
                .storeId(store.getId())
                .kakaoId(store.getKakaoId())
                .name(store.getName())
                .category(store.getCategory())
                .phone(store.getPhone())
                .address(store.getAddress().getAddress())
                .roadAddress(store.getAddress().getRoadAddress())
                .longitude(store.getGeoLocation().getLongitude())
                .latitude(store.getGeoLocation().getLatitude())
                .location(GeoJsonConverter.of(store.getGeoLocation().getLongitude(), store.getGeoLocation().getLatitude()))
                .placeUrl(store.getPlaceUrl())
                .status(true)
                .userId(store.getUserId())
                .build();
    }

}

File: src/main/java/com/zerozero/store/infrastructure/mongodb/Store.kt
package com.zerozero.store.infrastructure.mongodb

import com.zerozero.core.util.GeoJsonConverter
import org.springframework.data.mongodb.core.geo.GeoJsonPoint
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document(collection = "store")
class Store(
    val storeId: UUID,
    val kakaoId: String,
    val name: String,
    val category: String,
    val phone: String,
    val address: String,
    val roadAddress: String,
    val longitude: String,
    val latitude: String,
    val location: GeoJsonPoint,
    val placeUrl: String,
    val status: Boolean,
    val userId: UUID,
) {
    companion object {
        fun from(store: com.zerozero.store.domain.model.Store): Store {
            return Store(
                storeId = store.id,
                kakaoId = store.kakaoId,
                name = store.name,
                category = store.category,
                phone = store.phone,
                address = store.address.address,
                roadAddress = store.address.roadAddress,
                longitude = store.geoLocation.longitude,
                latitude = store.geoLocation.latitude,
                location = GeoJsonConverter.of(
                    store.geoLocation.longitude,
                    store.geoLocation.latitude
                ),
                placeUrl = store.placeUrl,
                status = true,
                userId = store.userId
            )
        }
    }
}
